src/renderer/components/SaveChangesButton.js

```diff
@@ -17,154 +17,95 @@
-  }
+  const { classes, successMessage } = props;
 
-  handleButtonClick = async (event) => {
-    this.setState(
-      {
-        inProgress: true,
-      },
-      async () => {
-        await this.props.onClick(event);
-        this.setState({
-          success: true,
-          inProgress: false,
-        });
-        this.timer = setTimeout(() => {
-          this.setState({ success: false });
-        }, 2000);
-      }
-    );
+  const handleButtonClick = async (event) => {
+    setInProgress(true);
+    console.log("about to do aysnc sutff");
+    console.log(" about to call  the onlick handler");
+    await props.onClick(event);
+    console.log("Got back from click");
+    setSuccess(true);
+    setInProgress(false);
+    setTimeout(() => {
+      console.log("running the timeout callback ");
+      setSuccess(false);
+      console.log("finishied timeout callback");
+    }, 2000);
   };
 
-  render() {
-    const { inProgress, success } = this.state;
-    const { classes, successMessage } = this.props;
-    const buttonClassname = classNames({
-      [classes.buttonSuccess]: success,
-    });
-
-    const textPart = !this.props.floating && (
-      <div className={classes.wrapper}>
-        <Button
-          variant="contained"
-          color="primary"
-          className={buttonClassname}
-          disabled={inProgress || (this.props.disabled && !success)}
-          onClick={this.handleButtonClick}
-        >
-          {success
-            ? successMessage || i18n.t("components.save.success")
-            : this.props.children}
-        </Button>
-      </div>
-    );
+  const textPart = !props.floating && (
+    <Box className={classes.wrapper}>
+      <Button
+        variant="contained"
+        disabled={inProgress || (props.disabled && !success)}
+        onClick={handleButtonClick}
+        color={success ? "success" : "primary"}
+      >
+        {success
+          ? successMessage || i18n.t("components.save.success")
+          : props.children}
+      </Button>
+    </Box>
+  );
 
-    const icon = this.props.icon || <SaveAltIcon />;
+  const icon = props.icon || <SaveAltIcon />;
 
-    const OptionalTooltip = (props) => {
-      if (this.props.floating) {
-        return <Tooltip title={this.props.children}>{props.children}</Tooltip>;
-      }
-      return props.children;
-    };
+  const OptionalTooltip = (props) => {
+    if (props.floating) {
+      return <Tooltip title={props.children}>{props.children}</Tooltip>;
+    }
+    return props.children;
+  };
 
-    return (
-      <OptionalTooltip>
-        <div
-          className={classNames(
-            classes.root,
-            this.props.className,
-            this.props.floating && classes.fab
+  return (
+    <OptionalTooltip>
+      <Box
+        sx={{
+          position: "fixed",
+          justifyContent: "flex-end",
+          bottom: 0,
+          left: 32,
+          zIndex: (theme) => theme.zIndex.drawer + 1,
+        }}
```

Comment:
These are properties we need to add *only* when `props.floating` is set.

We're using the `SaveChangesButton` both in the Editor, as a Floating Action Button, and at other places, like the `Upgrade` button on the FirmwareUpgrade screen, where it is a circular button + a normal button kinda mashed together.

We broke the opacity styling at some point, so they no longer look that well....

On second thought, see my global comment regarding this component. I'll post that in a minute.

Reply:
I think this is addressed now. I agree that eventually we should split this apart, but for now this should be ok.